### ISSUE TITLE
Article doesn't follow important event conventions

### ISSUE THREAD

**Issue by robloo** [dotnet/docs#13906]
Please implement the requirements from your other article https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/event in this one. This article does not follow a few important conventions. 

The problem starts in the first code snipped below. This has the name "On"EventRaised which should NOT be for the delegate "On" SHOULD be for the protected virtual method to raise each event.

```
void OnEventRaised(object sender, EventArgs args);
```

Instead the delegate should be something like:

```
void EventHandler(object sender, EventArgs args);
```

Then you need to add to this article the method for raising events "OnFileFound" and stop invoking FileFound directly. This is more than necessary if classes are ever derived and the whole framework more-or-less follows this convention.

Also, don't just add the method. Explain why it's added (for derived classes) and link to your other article on it that does things correctly: https://docs.microsoft.com/en-us/dotnet/standard/events/

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 90b37f1a-cd59-d9e5-067b-30ea5f26e5cc
* Version Independent ID: d2078536-1ee3-5b60-3872-f2635d245cc2
* Content: [Standard .NET event patterns](https://docs.microsoft.com/en-us/dotnet/csharp/event-pattern#feedback)
* Content Source: [docs/csharp/event-pattern.md](https://github.com/dotnet/docs/blob/master/docs/csharp/event-pattern.md)
* Product: **dotnet-csharp**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

**Comment by Thraka** [comment]
Thank you for opening this issue. I'll add it to the backlog. You can always submit the fix yourself by editing the original article. Click on the **Content Source** link at the bottom of your original comment. To learn how to edit, see the [Editing files in a repository](https://help.github.com/articles/editing-files-in-your-repository) article from GitHub.

Thanks again!

**Comment by robloo** [comment]
I think this needs to be re-opened. It was never closed and is still valid. I get that the backlog is getting long, but just forgetting things isn't the proper way to clear it out.

**Comment by IEvangelist** [comment]
Closed by #30994
